drv/gimlet-hf-server/src/main.rs

```diff
@@ -388,6 +395,34 @@ impl idl::InOrderHostFlashImpl for ServerImpl {
         poll_for_write_complete(&self.qspi);
         Ok(())
     }
+
+    fn get_mux(
+        &mut self,
+        _: &RecvMessage,
+    ) -> Result<HfMuxState, RequestError<HfError>> {
+        Ok(self.mux_state)
+    }
+
+    fn set_mux(
+        &mut self,
+        _: &RecvMessage,
+        state: HfMuxState,
+    ) -> Result<(), RequestError<HfError>> {
+        let gpio_driver = gpio_api::Gpio::from(GPIO.get_task_id());
+
+        let rv = match state {
+            HfMuxState::SP => gpio_driver.reset(self.select_pin),
+            HfMuxState::HostCPU => gpio_driver.set(self.select_pin),
```

Comment:
I need to change the gpio operations to not return Result, because they can only return client errors -- it hasn't been ported to Idol yet. But, consider just unwrapping here. If it fails, it's because `select_pin` is invalid for the _chip,_ which isn't something the caller can do anything about.